tracer/src/Datadog.Trace/AppSec/EventTrackingSdk.cs

```diff
@@ -17,172 +16,194 @@ namespace Datadog.Trace.AppSec;
+    }
+
+    /// <summary>
+    /// Sets the details of a logon failure on the local root span
+    /// </summary>
+    /// <param name="userId">The userId associated with the login failure</param>
+    /// <param name="exists">If the userId associated with the login failure exists</param>
+    /// <param name="metadata">Metadata associated with the login failure</param>
+    [PublicApi]
+    public static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata)
+    {
+        TelemetryFactory.Metrics.Record(PublicApiUsage.EventTrackingSdk_TrackUserLoginFailureEvent_Metadata);
+        TrackUserLoginFailureEvent(userId, exists, metadata, Tracer.Instance);
+    }
+
+    internal static void TrackUserLoginFailureEvent(string userId, bool exists, IDictionary<string, string> metadata, Tracer tracer)
+    {
+        if (string.IsNullOrEmpty(userId))
         {
-            if (string.IsNullOrEmpty(userId))
-            {
-                ThrowHelper.ThrowArgumentNullException(nameof(userId));
-            }
-
-            var span = tracer?.ActiveScope?.Span;
+            ThrowHelper.ThrowArgumentNullException(nameof(userId));
+        }
 
-            if (span is null)
-            {
-                ThrowHelper.ThrowException("Can't create a tracking event with no active span");
-            }
+        var span = tracer.ActiveScope?.Span;
 
-            var setTag = TaggingUtils.GetSpanSetter(span);
+        if (span is null)
+        {
+            ThrowHelper.ThrowException("Can't create a tracking event with no active span");
+        }
 
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureTrack, "true");
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureSdkSource, "true");
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureUserId, userId);
-            setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureUserExists, exists ? "true" : "false");
+        var setTag = TaggingUtils.GetSpanSetter(span, out var spanInternal);
 
-            if (metadata != null)
-            {
-                foreach (var kvp in metadata)
-                {
-                    setTag($"{Tags.AppSec.EventsUsers.LoginEvent.Failure}.{kvp.Key}", kvp.Value);
-                }
-            }
+        setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureTrack, "true");
+        setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureSdkSource, "true");
+        setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureUserId, userId);
+        setTag(Tags.AppSec.EventsUsers.LoginEvent.FailureUserExists, exists ? "true" : "false");
 
-            if (span is Span internalSpan)
+        if (metadata is { Count: > 0 })
+        {
+            var successTagPrefix = $"{Tags.AppSec.EventsUsers.LoginEvent.Failure}.";
+            foreach (var kvp in metadata)
             {
-                Security.Instance.SetTraceSamplingPriority(internalSpan);
+                setTag(successTagPrefix + kvp.Key, kvp.Value);
```

Comment:
Splitting might be less performant (depending on compiler optimisations) and I can't see a good reason to split it? 🤔

Reply:
thanks, fixed in [8fa918f](https://github.com/DataDog/dd-trace-dotnet/pull/6225/commits/8fa918f63e234e07613f17ed1634f12fdb342a0e)